common/src/main/java/com/revenuecat/purchases/common/networking/Endpoint.kt

```diff
@@ -0,0 +1,31 @@
+package com.revenuecat.purchases.common.networking
+
+import android.net.Uri
+
+sealed class Endpoint(val pathTemplate: String, val name: String) {
```

Comment:
I added a name, which will be used when posting data to the diagnostics endpoint. I thought about posting the path template but that wasn't very readable.

Reply:
Nice 👍🏻 
I did this refactor for iOS and it's been useful.